### ISSUE TITLE
Structure of `require.cache` is altered with ncc

### ISSUE THREAD

**Issue by amio** [zeit/ncc#243]
Given this `index.js`:
```js
console.log(require.cache);
```

Invoke with `node index.js`:
```
[Object: null prototype] {
  '/private/tmp/ncc/index.js':
   Module {
     id: '.',
     exports: {},
     parent: null,
     filename: '/private/tmp/ncc/index.js',
     loaded: false,
     children: [],
     paths:
      [ '/private/tmp/ncc/node_modules',
        '/private/tmp/node_modules',
        '/private/node_modules',
        '/node_modules' ] } }
```
with `ncc run index.js`:
```
2kB  index.js
2kB  [641ms]
{ '406': { i: 406, l: false, exports: {} } }
```

The structure is altered, which cause cache cleaning libs like `decache` or `clear-module` failed. Is this intentional or unexpected?

**Comment by guybedford** [comment]
There could be techniques to better support this certainly, but we would only be able to provide a very limited fake require.cache, just to ensure such packages don't fail entirely.

Do you have an example of `decache` or `clear-module` or a dependent of them that breaks here? It would help to understand the exact usage as to what is needed to support what workflow.

**Comment by amio** [comment]
Yeah, here's my testing code:
```js
// foo.js

let i = 0;
module.exports = function foo () { return ++i; };
```

```js
// index.js

const clearModule = require('clear-module');

logFoo(); //=> 1
logFoo(); //=> 2

clearModule(require.resolve('./foo'));

logFoo(); //=> 1

function logFoo () {
  console.log(require('./foo')());
}
```

**Comment by amio** [comment]
I think it's because they are deleting cache by file path, which is fine by require.cache's spec:
https://github.com/sindresorhus/clear-module/blob/d1faea712ffe8163c702492df23514f640043804/index.js#L25

It would be better adapted to `require.resolve(filePath)` other than `filePath`. This works for earlier example:

```js
// clearModule(require.resolve('./foo'));
delete require.cache[require.resolve('./foo')];
```

But in my real case, `require.resolve(fsPath)` returns the `[Function]` instead of the cache item's id. That's wired 🤔

**Comment by guybedford** [comment]
@amio ah, I thought you meant there was an error here in the `clearModule` execution, that we could patch to execute at least, but it seems that the issue is actually getting parity with the cache API.

Full cache API support isn't a natural feature of ncc due to the constraints of module optimization. There may be ways to approach this, but it's not something being prioritized.

**Comment by amio** [comment]
Yes I meant that :P, there is an error in `clearModule`'s execution.

If ncc could use same "id"(file path instead of a number) as require.cache, I guess it should make `clearModule` happy, and my use case too.
